### ISSUE TITLE
high severity in jsonwebtoken <=8.5.1

### ISSUE THREAD

**Issue by huineng** [IBM/node-sdk-core#222]
i got this warning this morning 

thanks

```
# npm audit report

jsonwebtoken  <=8.5.1
Severity: high
jsonwebtoken's insecure implementation of key retrieval function could lead to Forgeable Public/Private Tokens from RSA to HMAC - https://github.com/advisories/GHSA-hjrf-2m68-5959
jsonwebtoken vulnerable to signature validation bypass due to insecure default algorithm in jwt.verify() - https://github.com/advisories/GHSA-qwph-4952-7xr6
jsonwebtoken unrestricted key type could lead to legacy keys usage  - https://github.com/advisories/GHSA-8cf7-32gw-wr33
jsonwebtoken has insecure input validation in jwt.verify function - https://github.com/advisories/GHSA-27h2-hvpr-p74q
No fix available
node_modules/ibm-cloud-sdk-core/node_modules/jsonwebtoken
  ibm-cloud-sdk-core  >=0.3.0
  Depends on vulnerable versions of jsonwebtoken
  node_modules/ibm-cloud-sdk-core
    @ibm-cloud/cloudant  *
    Depends on vulnerable versions of ibm-cloud-sdk-core
    node_modules/@ibm-cloud/cloudant
```

**Comment by dpopp07** [comment]
@huineng Thanks for the notice. I am seeing this as well. It looks like a new major release of `jsonwebtoken` may be required. I'm not sure yet what changes we'll need in our code to support that but I will start looking into it

**Comment by sjcotto** [comment]
Hi, any news regarding this topic? it looks no major changes are needed

**Comment by TannerS** [comment]
Also wanted to check on this, due dates to fix deps which depend on this dep by januray 5th

**Comment by dpopp07** [comment]
Hey all - this will be resolved in #225

**Comment by davehorton** [comment]
what am I missing?  `npm audit` still shows this vulnerability in my project:

```
$ npm audit
# npm audit report

jsonwebtoken  <=8.5.1
Severity: moderate
jsonwebtoken's insecure implementation of key retrieval function could lead to Forgeable Public/Private Tokens from RSA to HMAC - https://github.com/advisories/GHSA-hjrf-2m68-5959
jsonwebtoken unrestricted key type could lead to legacy keys usage  - https://github.com/advisories/GHSA-8cf7-32gw-wr33
jsonwebtoken vulnerable to signature validation bypass due to insecure default algorithm in jwt.verify() - https://github.com/advisories/GHSA-qwph-4952-7xr6
fix available via `npm audit fix`
node_modules/jsonwebtoken
  ibm-cloud-sdk-core  0.3.0 - 3.2.2-beta-5 || 4.0.0
  Depends on vulnerable versions of jsonwebtoken
  node_modules/ibm-cloud-sdk-core

2 moderate severity vulnerabilities

To address all issues, run:
  npm audit fix

MacBook-Pro-3:realtimedb-helpers dhorton$ npm ls jsonwebtoken
@jambonz/realtimedb-helpers@0.6.3 /Users/dhorton/beachdog-enterprises/beachdog-networks/git/jambones.org/git/utils/realtimedb-helpers
└─┬ ibm-watson@7.1.2
  └─┬ ibm-cloud-sdk-core@3.2.1
    └── jsonwebtoken@8.5.1
```
